Don't use relative moment require paths

tofumatt opened the pull request:
The issue was looking for `../moment` and since we're already playing with the locales the easiest thing was to modify the `require` directly. This fixes #1628.

tofumatt commented:
d87b2d0bf800a4a694279ef67dfd0a224133aa1e is the commit you want to r? in this PR; the other commit is just the chore of rebuilding the locale files to have the right require path.

muffinresearch commented on webpack.dev.config.babel.js:
```diff
@@ -128,4 +126,15 @@ export default Object.assign({}, webpackConfig, {
     new webpack.IgnorePlugin(/webpack-stats\.json$/),
     webpackIsomorphicToolsPlugin.development(),
   ],
+  resolve: {
```
Is this not already there via the dev config being derived from the prod one?

tofumatt replied:
Actually yes, sorry this was a holder from testing and should be removed, thanks!

kumar303 commented on src/locale/id/amo.js:
```diff
@@ -20,32 +20,32 @@ module.exports = { domain:"messages",
-      "No ratings":[ "" ],
-      "%(users)s user":[ "" ],
-      "Log out":[ "" ],
+      "How are you enjoying your experience with %(addonName)s?":[ "Bagaimana pengalaman Anda menggunakan %(addonName)s?" ],
```
Why does this patch include translations? Did you mean to include these?

tofumatt replied:
Yeah, because the locale files were causing errors and needed to be rebuilt to fix them; I figured I'd include them here. Otherwise this won't show as fixed until we rebuilt the locales which won't be until next week.

tofumatt commented:
Yeah I'm not extremely happy with how we're doing this either but every JS i18n date library has locales that include code (for things like "1st, 2nd, 3rd" type numbers in different kinds of languages)... after a fair bit of messing about this was the best/most performant way to include it with our locales.

Definitely up for a better way if found though.

kumar303 commented:
Adding a note to my self: we chatted on IRC about auto-generating a require statement instead, which seems to work: https://dpaste.de/9v8u

tofumatt commented:
Huzzah, this is updated to not copy/paste the moment locale and instead require it.

I think the reason this used to not work in the first place is incredibly dumb oversight; I forgot to check for the locale in moment first and just added the require. So locales like `ast` (which moment doesn't yet support) would cause errors. Once I fixed that it worked and this is way better.

Thanks for the suggestion on IRC @kumar303! Ready for another r!

kumar303 commented on bin/build-locales:
```diff
@@ -50,15 +50,15 @@ poFiles.forEach((pofile) => {
       const momentLocale = locale.replace('_', '-').toLowerCase();
+      // Check for the locale first; if it doesn't exist we don't have
+      // a moment locale that matches.
+      fs.accessSync(`./node_modules/moment/locale/${momentLocale}.js`);
```
To make the access check more effective, you could share the same variable by doing:
````js
const localeModule = `moment/locale/${momentLocale}.js`;
fs.accessSync(`./node_modules/${localeModule}`);
// ...
defineLocale = new Function(`require('${localeModule}');`);
````

kumar303 commented on bin/build-locales:
```diff
@@ -50,15 +50,15 @@ poFiles.forEach((pofile) => {
-          `./node_modules/moment/locale/${momentLocale}.js`, 'utf8'
-        )
-      );
+        `require('moment/locale/${momentLocale}.js');`);
```
do you still need `new Function()` ?

kumar303 replied:
Since requiring this module is magical, I think it would make sense to add a comment to the auto-generated code like:

    // By requiring this module, the new locale is defined and registered internally for moment.

tofumatt replied:
Yeah, because if we include `require` in there directly it will evaluate and throw an error, and if we make it a string `toSource` will output it as one. It's now quite harmless; a bit ugly but that's all.

tofumatt replied:
Works for me! 👍

tofumatt commented:
Changes made (and bit about `new Function` usage explained). I think this should be good now, r?

kumar303 commented on bin/build-locales:
```diff
@@ -50,15 +50,18 @@ poFiles.forEach((pofile) => {
+      defineLocale = new Function(`
+        // By requiring this module, the new locale is defined and
+        // registered internally for moment.js
+        require('${localeModulePath}');`);
```
I guess it's ok that the module path ends in `.js` now? That's probably fine.

tofumatt replied:
Good point; I can remove it though, no need for it.

tofumatt commented:
(I forgot I didn't merge this.)
